### PR TITLE
Remove default batch params if submitting to HQS family device.

### PR DESCRIPTION
**Pull request by vanyae-cqc**
HQS (and pytket-honeywell) now allow users to submit to a 'device family' ("HQS-LT") instead of specifying the exact device they want to run on.

We checked with the developers of the HQS API and it turns out that if jobs are submitted to a device family and batching is requested in the request JSON then an error will be returned (as the circuits may be run on different machines).

This PR removes the insertion of default batch params in the case that the user is submitting to a device family. If a user manually inserts batch params then they the API will still return an error, which I figured should be the correct behaviour.

### CODE REVIEW

File: modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py [hunk 1]
```diff
@@ -363,10 +364,13 @@ def process_circuits(
 
             if final_index > 0:
                 # only set batch fields if more than one job submitted
-                body["batch-exec"] = batch_exec
-                if i == final_index:
-                    # flag to signal end of batch
-                    body["batch-end"] = True
+                if (self._device_name != DEVICE_FAMILY) or ("max_batch_cost" in kwargs):
```
Comment (ss2165) [inline modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py:367]: The only suggestion I would make is `and`ing this condition with the `if final_index >0` so there is a single decision point as to whether to batch or not.

**Review by ss2165 — DISMISSED** [review]
Just a minor suggestion, looks good otherwise.

File: modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py (hunk 1 above)
Comment (vanyae-cqc) [reply modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py:367]: Yeah that makes sense, will do!

**Comment by vanyae-cqc** [comment]
Merged my fork with Alec's recent merge. He wasn't sure why the MacOs action timed out but suggested running it again. Fingers crossed!

**Pull request merged**